leptos_dom/src/lib.rs

```diff
@@ -775,6 +775,67 @@ impl View {
 
         self
     }
+
+    /// Adds a directive which is a function that is run on mounted. A directive can be a function
+    /// with one or two parameters. The first is the element the directive is added to and the optional
+    /// second is the paramter that is provided in the attribute.
+    pub fn directive<T, P>(
+        self,
+        handler: impl DirectiveHandler<T, P>,
+        param: &P,
+    ) -> Self {
+        cfg_if! { if #[cfg(all(target_arch = "wasm32", feature = "web"))] {
+            match &self {
+                Self::Element(el) => {
+                    let _ = el.element.on_mount(move |el| handler.call(el, param));
```

Comment:
Note that `on_mount` will only ever run once, so if the element is unmounted and remounted, or moved, it won't run again. This may or may not be desirable, I'm not sure.

Reply:
Yeah I read that in the docs comment, but I wasn't sure either 😄